test/lib/cql_test_env.cc

```diff
@@ -479,7 +479,11 @@ class single_node_cql_env : public cql_test_env {
             if (!cfg->view_update_reader_concurrency_semaphore_kill_limit_multiplier.is_set()) {
                 cfg->view_update_reader_concurrency_semaphore_kill_limit_multiplier.set(std::numeric_limits<uint32_t>::max());
             }
-            tmpdir data_dir;
+            fs::path temp_dir_path = fs::temp_directory_path();
+            if (init_configurables.tmp_dir_path) {
+                temp_dir_path = *init_configurables.tmp_dir_path;
+            }
+            tmpdir data_dir(temp_dir_path);
```

Comment:
Ok. This answers my top-level comment.

Will it be deleted on a clean exit?

How much space is consumed?

It's better to have an in-memory environment, but this is acceptable too.